expression/builtin_encryption_vec.go

```diff
@@ -139,11 +144,79 @@ func (b *builtinMD5Sig) vecEvalString(input *chunk.Chunk, result *chunk.Column)
 }
 
 func (b *builtinSHA2Sig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinSHA2Sig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf1); err != nil {
+		return err
+	}
+	result.ReserveString(n)
+	i64s := buf1.Int64s()
+	buf.MergeNulls(buf1)
+	var (
+		hasher224 hash.Hash
+		hasher256 hash.Hash
+		hasher384 hash.Hash
+		hasher512 hash.Hash
+	)
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		hashLength := i64s[i]
+		var hasher *hash.Hash
```

Comment:
```suggestion
		var hasher hash.Hash
```

We don't have to use a pointer of an interface here. The original value of an interface we state here is null.
See the example https://play.golang.org/p/r1GECExUiRj .

Reply:
I will fix it.